homeassistant/components/plaato/config_flow.py

```diff
@@ -1,10 +1,227 @@
+            data=self._init_info,
+            description_placeholders={
+                PLACEHOLDER_DEVICE_TYPE: device_type.name,
+                PLACEHOLDER_DEVICE_NAME: device_name,
+            },
+        )
+
+    async def _show_api_method_form(self, device_type: PlaatoDeviceType):
+        data_scheme = vol.Schema({vol.Optional(CONF_TOKEN, default=""): str})
+
+        if device_type == PlaatoDeviceType.Airlock:
+            data_scheme = data_scheme.extend(
+                {vol.Optional(CONF_USE_WEBHOOK, default=False): bool}
+            )
+
+        return self.async_show_form(
+            step_id="api_method",
+            data_schema=data_scheme,
+            errors=self._errors,
+            description_placeholders={PLACEHOLDER_DEVICE_TYPE: device_type.name},
+        )
+
+    async def _get_webhook_id(self):
+        """Generate webhook ID."""
+        webhook_id = self.hass.components.webhook.async_generate_id()
+        if self.hass.components.cloud.async_active_subscription():
+            webhook_url = await self.hass.components.cloud.async_create_cloudhook(
+                webhook_id
+            )
+            cloudhook = True
+        else:
+            webhook_url = self.hass.components.webhook.async_generate_url(webhook_id)
+            cloudhook = False
+
+        return webhook_id, webhook_url, cloudhook
+
+    @staticmethod
+    def _get_error(device_type: PlaatoDeviceType):
+        return (
+            "no_api_method"
+            if device_type == PlaatoDeviceType.Airlock
+            else "no_auth_token"
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return PlaatoOptionsFlowHandler(config_entry)
+
+
+class PlaatoOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Plaato options."""
+
+    def __init__(self, config_entry: ConfigEntry):
+        """Initialize domain options flow."""
+        super().__init__()
+
+        self._config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        use_webhook = self._config_entry.data.get(CONF_USE_WEBHOOK, False)
+        if use_webhook:
+            return await self.async_step_webhook()
+
+        return await self.async_step_user()
+
+    async def async_step_user(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
+                        default=self._config_entry.options.get(
+                            CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                        ),
+                    ): cv.positive_int
+                }
+            ),
+        )
```

Comment:
Does this need to be configurable?  As developers we should choose a reasonable default (lowest interval that is reasonable) and not make it configurable unless there is a strong use case where the device will have issues if its not adjustable.

Reply:
I received request on being able to configure this from users, hence I thought I would honor their request. I think it is a good idea since depending on if you monitor a Keg or a Airlock you would want to haver different intervals.